cloudinit/config/cc_set_hostname.py

```diff
@@ -21,9 +21,13 @@
 the ``fqdn`` config key. If both ``fqdn`` and ``hostname`` are set, ``fqdn``
 will be used.
 
+In order to correctly set the hostname, cloud-init will always run this module
+during the init stage. This ensures that the hostname is correctly set on
+some datasources before a DHCP lease is given.
+
 **Internal name:** ``cc_set_hostname``
 
-**Module frequency:** per instance
+**Module frequency:** per always
```

Comment:
"per always" sounds funny to me, but I see we use it in other modules, so maybe I'm just not familiar with the expression.

Reply:
I agree I originally thought "per boot" would be what it was, but apparently not!